allennlp/training/metrics/pearson_correlation.py

```diff
@@ -69,9 +75,21 @@ def get_metric(self, reset: bool = False):
         covariance = self._predictions_labels_covariance.get_metric(reset=reset)
         predictions_variance = self._predictions_variance.get_metric(reset=reset)
         labels_variance = self._labels_variance.get_metric(reset=reset)
+        denominator = math.sqrt(predictions_variance) * math.sqrt(labels_variance)
+        if is_distributed():
+            # Note: this gives an approximate aggregation of the covariance.
+
+            # Aggregating on the CPU since these are single float values.
+            device = torch.device("cpu")
```

Comment:
@epwalsh mentioned that this won't work when the backend is NCCL though?

Reply:
Yes, correct. Fixing.